src/libraries/Microsoft.PowerFx.Interpreter/Environment/PowerFxConfigExtensions.cs

```diff
@@ -45,5 +45,21 @@ public static void EnableMutationFunctions(this SymbolTable symbolTable)
             symbolTable.AddFunction(new ClearFunction());
             symbolTable.AddFunction(new ClearCollectFunction());
         }
+
+        public static void EnableRegExFunctions(this PowerFxConfig config, TimeSpan regExTimeout = default, int regexCacheSize = -1)
+        {
+            ConcurrentDictionary<string, Tuple<DType, bool, bool, bool>> regexTypeCache = regexCacheSize == -1 ? null : new ConcurrentDictionary<string, Tuple<DType, bool, bool, bool>>();
```

Comment:
>ConcurrentDictionary<string, Tuple<DType, bool, bool, bool>> [](http://example.com/codeflow?start=12&length=60)

what is this type?

Reply:
ConcurrentDictionary<string, Tuple<DType, bool, bool, bool>> :)

Coming from PA Client code